WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/wizards/RegistrationWizard.java

```diff
@@ -0,0 +1,76 @@
+package nl.tudelft.watchdog.ui.wizards;
+
+import nl.tudelft.watchdog.ui.preferences.Preferences;
+import nl.tudelft.watchdog.ui.util.UIUtils;
+import nl.tudelft.watchdog.ui.wizards.projectregistration.ProjectCreatedEndingPage;
+import nl.tudelft.watchdog.ui.wizards.projectregistration.ProjectIdEnteredEndingPage;
+import nl.tudelft.watchdog.ui.wizards.projectregistration.ProjectRegistrationPage;
+import nl.tudelft.watchdog.ui.wizards.projectregistration.ProjectSliderPage;
+import nl.tudelft.watchdog.ui.wizards.projectregistration.ProjectWelcomePage;
+
+import org.eclipse.jface.wizard.Wizard;
+
+/** Parent class for User and Project registration wizards. */
+public abstract class RegistrationWizard extends Wizard {
+
+	/**
+	 * The project-id, either entered on the previous wizard pages or as
+	 * retrieved by the server.
+	 */
+	protected String projectId;
+
+	/** Maximum number of pages in the current wizard. */
+	protected int totalPageNumber;
+
+	/** Page with Yes/No question about registering new project ID. */
+	public ProjectWelcomePage projectWelcomePage;
```

Comment:
I am currently also not sure why the wizard base should no about the specific projectWelcomePage. Isn't this a logical flaw?